doc/users.md

```diff
@@ -1 +1,1712 @@
+
+Response Payload :
+```
+{
+    "status_code": "cdc-200",
+    "status_message": "no change is requested",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "cdc-200",
+    "status_message": "no change is requested",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "name": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "name is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload
+```
+{
+    "type_akses": 9
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "type_akses must in range 1-8",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+Request Payload
+```
+{
+    "type_akses": 1
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "company_id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 6
+
+Request Payload
+```
+{
+    "type_akses": 1,
+    "company_id": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "company_id is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 7
+
+Request Payload
+```
+{
+    "type_akses": 1,
+    "company_id": "test"
+}
+```
+
+Response HTTP Status Code : 404
+
+Response Payload
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "company_id not found in companies",
+    "data": null
+}
+```
+
+#### Case : Negative Case 8
+
+Request Payload
+```
+{
+    "type_akses": 2,
+    "company_id": "5fb5df66-5f80-416d-a055-401bcf944ef7"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "region_id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 9
+
+Request Payload
+```
+{
+    "type_akses": 2,
+    "company_id": "5fb5df66-5f80-416d-a055-401bcf944ef7",
+    "region_id": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "region_id is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 10
+
+Request Payload
+```
+{
+    "type_akses": 2,
+    "company_id": "5fb5df66-5f80-416d-a055-401bcf944ef7",
+    "region_id": "test"
+}
+```
+
+Response HTTP Status Code : 404
+
+Response Payload
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "region_id not found in regions",
+    "data": null
+}
+```
+
+#### Case : Negative Case 11
+
+Request Payload
+```
+{
+    "type_akses": 3,
+    "company_id": "5fb5df66-5f80-416d-a055-401bcf944ef7"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "branch_id is required",
```

Comment:
in edit mode, if there is no branch_id, you can use the existing data. it's mean the branch_id is not changed. But if any update of company_id, you should revalidate : "whether the existing branch_id is still belongs to company_id"?

also you should revalidate if any change region_id : "whether the existing branch_id is still belongs to region_id"?